clients/ts/signalr/tests/HubConnection.test.ts

```diff
@@ -163,12 +163,14 @@ describe("HubConnection", () => {
                     protocolCalled = true;
                 };
 
-                const connection = new TestConnection();
+                const connection = new TestConnection(false);
                 const hubConnection = createHubConnection(connection, logger, mockProtocol);
                 try {
+                    const startPromise = hubConnection.start();
                     const data = "{}" + TextMessageFormat.RecordSeparator;
 
                     connection.receiveText(data);
+                    await startPromise;
```

Comment:
For the tests that are directly about the handshake, I'd like to assert that the promise isn't resolved before the `receiveText` method is called.

You can do that with code like this:

```javascript
let startCompleted = false;
const startPromise = hubConnection.start().then(() => startCompleted = true);
...
expect(startCompleted).toBe(false);
connection.receiveText(data);
await startPromise;
```